lua/starfall/permissions/core.lua

```diff
@@ -64,13 +64,27 @@ end
 -- @param instance The instance checking permission
 -- @param target the object on which the action is being performed
 -- @param key a string identifying the action being performed
-function P.check(instance, target, key)
+-- @return boolean Whether the player may perform the action
+-- @return string? Reason for action to not be allowed, or nil if it is allowed
+function P.checkSafe(instance, target, key)
 	if not (instance.permissionOverrides and instance.permissionOverrides[key]) then
 		local notok, reason = P.permissionchecks[key](instance, target)
 		if notok then
-			SF.Throw("Permission " .. key .. ": " .. reason, 3)
+			return false, reason
```

Comment:
What's the reason for removing the original error method?

Reply:
So that I can safely do a permission check from inside a hook. The new behavior is only in `CheckSafe`; `Check` is redefined to have the exact same behavior as before, so this change should not break anything.

Reply:
yea that was cleared up in the conversation happening in the discord. Our conclusion was this should be its own fork, an addon or its own extension.

Reply:
Doesn't the hasAccess function already do this?

Reply:
...huh. It does. Didn't even know that existed to be honest.